### ISSUE TITLE
Rename CSC primary branch to follow best practices

### ISSUE THREAD

**Issue by peterrhysthomas** [finos/cloud-service-certification#119]
CSC implementation for overall issue https://github.com/finos/community/issues/114

**Comment by TLATER** [comment]
Hi @mcleo-d, just a friendly reminder to figure out why I cannot be assigned ;)

**Comment by mcleo-d** [comment]
Hi @TLATER - I have created the Cloud Service Certification team in FINOS and have added you as a member. That means you can now be assigned to issues - https://github.com/orgs/finos/teams/cloud-cert-participants 👍 

Thanks for the reminder! 😸

**Comment by TLATER** [comment]
I've tested this here: https://github.com/TLATER/master-migration-test

GitHub is quite helpful and automatically changes the PR targets, but local "main/master" branches on contributor machines do need to be updated manually. Any local "upstream" targets for non-main branches will need to be updated too, for convenient diffing.

So we'll break contributors' local repositories a little. This seems fine as long as everybody is made aware.

The actual change is easy enough, someone with sufficient repository ownership just needs to change the name of the default branch in `Settings -> Branches`.

Other potential bits of fallout are in CI (if any CI system runs `git checkout https://github.com/finos/cloud-service-certification -b master` or such), but this isn't present in our repository yet, so there should be no other problems.

**Comment by TLATER** [comment]
I'll get some docs ready on how to update local branches, and ping them on the slack; we can then follow up in the next group meeting, and pull the trigger after that?

**Comment by TLATER** [comment]
Here's a script that will fix branch upstreams locally; note this makes use of bash-style string manipulation, so fish users will probably need something else (zsh works fine):

*Please run this **after** the main branch exists, not before*

```bash
git for-each-ref --format='%(refname:short).%(upstream:short)' refs/heads | \
    while read entry; do
        branch="${entry%.*}"
        upstream="${entry#*.}"

        # Changes the upstream branches of anything that was
        # previously pointing to master to main
        if [ "${upstream}" = "origin/master" ]; then
            echo git branch "${branch}" --set-upstream-to "origin/main"
        elif [ "${upstream}" = "master" ]; then
            echo git branch "${branch}" --set-upstream-to "main"
        fi

        # Sets the "default" origin branch to main
        git symbolic-ref refs/remotes/origin/HEAD refs/remotes/origin/main

        # Renames our local master branch to main as well
        git branch --move master main
    done
```

Any private forks will of course be unaffected by this, for those people will need to modify the branch names themselves. To fix branches for private forks:

```bash
REMOTE="<your-remote-name>"

git for-each-ref --format='%(refname:short).%(upstream:short)' refs/heads | \
    while read entry; do
        branch="${entry%.*}"
        upstream="${entry#*.}"

        # Changes the upstream branches of anything that was
        # previously pointing to master to main
        if [ "${upstream}" = "${REMOTE}/master" ]; then
            echo git branch "${branch}" --set-upstream-to "${REMOTE}/main"
        fi

        # Sets the "default" origin branch to main
        git symbolic-ref "refs/remotes/${REMOTE}/HEAD" "refs/remotes/${REMOTE}/main"
    done
```

It's probably easier to delete personal forks and re-create them, though.

**Comment by mcleo-d** [comment]
@maoo has been invited to discuss in the next CSC sprint meeting on 3rd June here ...
- https://github.com/finos/cloud-service-certification/issues/126#issuecomment-849534888

**Comment by mcleo-d** [comment]
Hi @TLATER (and @maoo)

I have extended CSC Sprint 2 until 17th June '21 due to team member annual leave and have added this item to the #127 agenda to discuss in relation to completion and Sprint 3.

Please find below a link to the CSC Channel in FINOS Slack where the following question was asked.
https://finos-lf.slack.com/archives/C01RNUMC2DP/p1622730250024900
  
![Screenshot 2021-06-11 at 12 00 42](https://user-images.githubusercontent.com/6029572/121676563-aec0f900-caac-11eb-918a-867b82e44395.png)

Maybe we can extend the vote and decide on the call for Sprint 3?

Cheers,

James 🚀

**Comment by mcleo-d** [comment]
Hi @maoo 👋🏻 

It was agreed during Cloud Service Certification meeting #127 that this is the top issue to resolve during Sprint 3 (https://github.com/orgs/finos/projects/1#column-14492299) and no pull requests will be raised or updated until done.

Can you work with @TLATER to update the `main` repo branch so @TLATER can make changes to underlying dependencies and document what other FINOS projects should do to follow?

Many thanks,

James 🚀

**Comment by mcleo-d** [comment]
Hi @TLATER 👋🏻 

Regarding the documentation for this change, it would be great if you could raise a PR adding markdown to the `development-infrastructure` section of the FINOS Open Developer Platform microsite ... https://github.com/finos/open-developer-platform/tree/master/docs/development-infrastructure

The `development-infrastructure` folder above ☝🏻 drives the development-infrastructure content found on the ODP site here https://odp.finos.org/docs/what-is-odp/. The microsite will be rebuilt when the PR is merged. 

Then, we can add a reference to your documentation to the [FINOS Maintainers Cheatsheet](https://odp.finos.org/docs/finos-maintainers-cheatsheet) where the content of the page is found here ... https://github.com/finos/open-developer-platform/blob/master/docs/finos-maintainer-cheatsheet.mdx. This is how we scale the change ... 🏎️ 💨 

When done, we can present on the FINOS All Community Call at the end of July. WDYT?

James 🚀

**Comment by TLATER** [comment]
Yes, this sounds good, I'll get a PR up there asap :)

**Comment by TLATER** [comment]
I think we're blocked by the above PR after all - absolutely my bad for forgetting to check for absolute links in markdown.

**Comment by mcleo-d** [comment]
This is on hold until CSC Sprint 4 kickoff to unblock @leefaus and @J0EG on their merging and raising of pull requests ...

**Comment by mcleo-d** [comment]
@TLATER and @maoo 👋🏻 

This is to confirm this issue has been prioritised at the top of Sprint 4 that ends of 12th August.

It would be great if this item can be completed ASAP.

James.

**Comment by mcleo-d** [comment]
@TLATER and @peterrhysthomas 

The Cloud Service Certification `master` branch is now renamed to `main`! Please notify the CSC team with the following as CSC members will have to manually update their local environments:

```
git branch -m master main
git fetch origin
git branch -u origin/main main
```

**Comment by mcleo-d** [comment]
The following PR has been raised to add the update to the repo README.md #132 

@TLATER and @peterrhysthomas can you review and accept the PR above ☝🏻 

Thanks!

James 🚀
